src/v/security/gssapi_authenticator.cc

```diff
@@ -494,8 +501,13 @@ gssapi_authenticator::impl::check() {
     //     return errc::invalid_credentials;
     // }
 
+    _rp_audit_user.credential_uid = ss::sstring{source_name};
+
     _rp_user_principal = get_principal_from_name(source_name);
 
+    _rp_audit_user.name = _rp_user_principal.name();
+    _rp_audit_user.type_id = audit::user::type::user;
```

Comment:
> I think _rp_audit_user can be set inside get_principal_from_name, where the context is less disjointed.

👍 

> It appears as though credential_uid is set twice, possible to different values.

Yes this was intentional.  The first time it's set, it's set to the unparsed value of the kerberos principal.  This is in case parsing on it fails, and therefor authentication fails.  The audit log would then contain this invalid principal so the auditor could attempt to analyze why authn failed.

However after thinking about it, the second time it's assigned is probably unnecessary as the parsed value should match the originally provided kerberos principal.  Will remove the second assignment.